executor/executor_runner.h

```diff
@@ -621,11 +631,35 @@ class Runner
 				failmsg("unknown host message type", "type=%d", static_cast<int>(raw.msg.type));
 		}
 
+#if GOOS_linux
+		const uint64 kRunLeakCheckEvery = 2 * procs_.size();
+		constexpr auto IsProcIdle = [](const std::unique_ptr<Proc>& proc) {
+			return proc->GetState() == Proc::State::Idle;
+		};
+		const bool isScheduledForLeakCheck =
+		    is_leak_enabled_ &&
+		    corpus_triaged_ &&
+		    exec_count_ > 0 &&
+		    exec_count_ % kRunLeakCheckEvery == 0;
+
+		if (isScheduledForLeakCheck && std::all_of(procs_.begin(), procs_.end(), IsProcIdle)) {
+			const int len = leak_frames_.size();
+			auto leakFrames = GetArrOfCStr(leak_frames_);
+			check_leaks(leakFrames, len);
+			FreeArrOfCStr(leakFrames, len);
+			++exec_count_;
```

Comment:
I understand why you need to increase `exec_count_` here, but then it's not exactly an exec count either. Maybe do similar to `freshness_` then? Let it increase until it hits kRunLeakCheckEvery, then do the leak check and set it to 0. 
(We'll then need to rename it once more, of course).